src/main/java/com/redhat/devtools/lsp4ij/usages/LSPUsageSearcher.java

```diff
@@ -98,7 +115,12 @@ public void processElementUsages(@NotNull PsiElement element, @NotNull Processor
         }
 
         // For completeness' sake, also collect external usages to LSP (pseudo-)elements
-        LSPExternalReferencesFinder.processExternalReferences(file, element.getTextOffset(), reference -> processor.process(new UsageInfo2UsageAdapter(new UsageInfo(reference))));
+        LSPExternalReferencesFinder.processExternalReferences(
+                file,
+                ReadAction.compute(element::getTextOffset),
+                options.searchScope,
+                reference -> processor.process(new UsageInfo2UsageAdapter(new UsageInfo(reference)))

```

Comment:
If I remember you can have a ReadAction problem here. (See the first call process.process which compute a ReadAction

Reply:
Do you mean that `new UsageInfo2UsageAdapter(new UsageInfo(reference))` should be enclosed in a `ReadAction` here like in the other similar constructor calls in the same method. I don't _think_ that's necessary because the `UsageInfo2UsageAdapter` constructor already includes its own `ReadAction`, and the `UsageInfo` constructor also looks fine. Having said that, I'm totally fine wrapping that in a `ReadAction` both for consistency and out of a sense of caution if you'd like. Let me know.

Reply:
I went ahead and added it.

Reply:
Yes it is that, I don't remember how to reproduce it, but I had to add ReadAction#compute to fix the problem. See https://github.com/redhat-developer/lsp4ij/blob/1783e367a3190af823c6f8a5766881133fef3d67/src/main/java/com/redhat/devtools/lsp4ij/usages/LSPUsageSearcher.java#L62